torch/futures/__init__.py

```diff
@@ -82,20 +148,19 @@ def then(self, callback):  # type: (Callable[[Future[T]], S]) -> Future[S]
         return cast(Future[S], super().then(callback))
 
     # Have to use string annotations because  PEP-0563 is not available in 3.6
-    def _add_done_callback(self, callback):  # type: (Callable[[Future[T]], None]) -> None
+    def add_done_callback(self, callback):  # type: (Callable[[Future[T]], None]) -> None
```

Comment:
I am removing the leading underscore from this method because, if it was intentional, I don't think it was really achieving the desired effect. It probably was intended to make this method "private", which it did, but it failed to override the `add_done_callback` method from the superclass. This method was appearing in the docs, however without any documentation (see https://pytorch.org/docs/master/futures.html#torch.futures.Future.add_done_callback). Either we somehow remove/hide the superclass's method, or we keep it but document it. I went for the latter.

I looked for usages of the underscore-prefixed version of this method in the codebase but couldn't find any.

Reply:
Agree. Leaving `add_done_callback` undocumented wasn't intentional, and we should fix. Both removing this method and making it public look OK to. There was a request from @jjlilley for this feature, and this also complies with Python APIs. There was also an argument for not exposing it, and `then` already covers a superset of use cases.

Reply:
this is actually necessary in use cases like the one below, right? 

https://discuss.pytorch.org/t/pytorch-distributed-rpc-bottleneck-in-recursive-compile-class/117005/5